### ISSUE TITLE
Protocol Style

### ISSUE THREAD

**Issue by shuaib88** [uchicago-mobi/2016-Winter-Forum#58]
Ray wunderlich suggests declaring protocols and keeping methods separate like this. Have you guys seen this often in industry or would this look weird? 

![screen shot 2016-01-27 at 11 48 07 am](https://cloud.githubusercontent.com/assets/9851472/12622636/ebe08ac0-c4eb-11e5-9a03-1ee8ae58a0ad.png)

https://github.com/raywenderlich/swift-style-guide#protocol-conformance

**Comment by tabinks** [comment]
This is a personal preference, there is no difference in the behavior.  The one advantage of breaking them out is to provide a visual distinction.  There is a well know problem of "Massive View Controllers" where view controllers have a tremendous amount of code.  [Let's Play 'Refactor the Mega-Controller' with Andy Matuschak - Realm is a mobile database: a replacement for SQLite & Core Data](https://realm.io/news/andy-matuschak-refactor-mega-controller/)

If you implement it this way (above), you can break it up your code so it is more manageable.  You can even put the extension in different files.  The one bad thing about doing this is that you will constantly be referring back to the view controller file while working on your extensions.

Also, remember that all the methods and properties of the view controller are available in the extension.  For example, a variable named `issues` that is declared in the view controller can be access as `issues` in the extension.

In summary, it's a point of style and, as far as I know, there is no general consensus on this.  Let's ask some experts: @uchicago-mobi/alumni Anyone seeing this in practice in 'the real world'?

**Comment by jhergott** [comment]
We break up our view controllers with `extensions` at the company I'm currently working for.  I agree with @tabinks :+1:, it's more or less just a way to modularize your code.  That being said there are a couple of cool things you can do with it.
-  You can mark an extension `private extension` and then all the `func`'s inside the extension will be private without explicitly calling `private func`.
- It can also be used as a way to add some swift code to an obj-c class.  Let's say you have an obj-c class `MyPerson.h/MyPerson.m`.  You can create a swift file and extend your obj-c class with `extension MyPerson`.

**Comment by yiqin** [comment]
Honestly, it's my first time to learn about it. However, what Ray wunderlich suggests is very reasonable. I hear about many developers complaint that the view controllers are doing too many things in traditional MVC. Breaking up view controllers with extensions is a good way to solve it. I'm going to write view controllers in this way next time.

**Comment by JRam13** [comment]
I would say something about MVVM but...

**Comment by tabinks** [comment]
@jhergott I was unaware of those.  Thanks for the tips.

**Comment by shuaib88** [comment]
very interesting to hear all those perspectives -- it seems like a good way to organize code without changing any of the functionality. Thanks guys!
